yarn-project/stdlib/src/abi/encoder.ts

```diff
@@ -142,6 +147,58 @@ class ArgumentEncoder {
     }
     return this.flattened;
   }
+
+  /**
+   * Encodes an array as a BoundedVec struct.
+   * @dev BoundedVec is handled as a special case rather than a generic struct for two reasons:
+   * 1. It is a commonly used type
+   * 2. Manual encoding it is cumbersome
+   * Therefore, the input is simplified to accept a plain array of type T.
+   * @param abiType - The ABI type definition.
+   * @param arg - An array of items to encode.
+   *
+   * The BoundedVec struct is defined in Noir as:
+   *
+   * ```noir
+   * pub struct BoundedVec<T, let MaxLen: u32> {
+   *   storage: [T; MaxLen],
+   *   len: u32,
+   * }
+   * ```
+   *
+   * The encoding follows Noir's serialization format:
+   * 1. The storage array is encoded first
+   * 2. The length field is encoded second
+   */
+  #encodeBoundedVec(abiType: AbiType, arg: any) {
+    // First we encode the storage array
+    {
+      // Get the storage array type from the BoundedVec struct
+      const storageField = (abiType as unknown as any).fields.find((f: any) => f.name === 'storage')!;
+      const maxLength = storageField.type.length;
```

Comment:
If `arg.length` is larger than this we need to throw an error, ideally a very nice looking one since getting this error will _suck_ (i.e. "error encoding param 'foo' of function 'bar': expected an array of maximum length X and got Y instead: [ arr_0, arr_1, ... ]").

Reply:
(also add a test please :upside_down_face:)

Reply:
Sorry, conveniently forgot the tests lol.

Addressed in https://github.com/AztecProtocol/aztec-packages/pull/14891/commits/83e1cbd0b125a3ad9ee8db1c02817c871d9033c9